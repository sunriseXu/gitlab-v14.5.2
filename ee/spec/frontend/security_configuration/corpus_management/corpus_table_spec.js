import { mount } from '@vue/test-utils';
import Actions from 'ee/security_configuration/corpus_management/components/columns/actions.vue';
import CorpusTable from 'ee/security_configuration/corpus_management/components/corpus_table.vue';
import { corpuses } from './mock_data';

const TEST_PROJECT_FULL_PATH = '/namespace/project';

describe('Corpus table', () => {
  let wrapper;

  const createComponentFactory = () => (options = {}) => {
    const defaultProps = {
      corpuses,
    };

    const defaultMocks = {
      $apollo: {
        mutate: jest.fn().mockResolvedValue(),
      },
    };

    wrapper = mount(CorpusTable, {
      propsData: defaultProps,
      provide: {
        projectFullPath: TEST_PROJECT_FULL_PATH,
      },
      mocks: defaultMocks,
      ...options,
    });
  };

  const createComponent = createComponentFactory();

  afterEach(() => {
    wrapper.destroy();
  });

  describe('corpus management', () => {
    beforeEach(() => {
      createComponent();
    });

    it('bootstraps and renders the component', () => {
      expect(wrapper.findComponent(CorpusTable).exists()).toBe(true);
    });

    it('renders with the correct columns', () => {
      const columnHeaders = wrapper.findComponent(CorpusTable).find('thead tr');
      expect(columnHeaders.element).toMatchSnapshot();
    });

    it('triggers the corpus deletion mutation', () => {
      const {
        $apollo: { mutate },
      } = wrapper.vm;

      const actionComponent = wrapper.findComponent(Actions);

      expect(mutate).not.toHaveBeenCalled();
      actionComponent.vm.$emit('delete', 'corpus-name');
      expect(mutate).toHaveBeenCalledTimes(1);
    });
  });
});
