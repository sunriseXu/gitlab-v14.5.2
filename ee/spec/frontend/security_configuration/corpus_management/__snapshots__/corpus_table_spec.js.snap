// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Corpus table corpus management renders with the correct columns 1`] = `
<tr
  class=""
  role="row"
>
  <th
    aria-colindex="1"
    class="gl-bg-transparent! gl-border-gray-100! gl-border-b-solid! gl-border-b-1!"
    role="columnheader"
    scope="col"
  >
    <div>
      Corpus name
    </div>
  </th>
  <th
    aria-colindex="2"
    class="gl-bg-transparent! gl-border-gray-100! gl-border-b-solid! gl-border-b-1!"
    role="columnheader"
    scope="col"
  >
    <div>
      Target
    </div>
  </th>
  <th
    aria-colindex="3"
    class="gl-bg-transparent! gl-border-gray-100! gl-border-b-solid! gl-border-b-1!"
    role="columnheader"
    scope="col"
  >
    <div>
      Last updated
    </div>
  </th>
  <th
    aria-colindex="4"
    class="gl-bg-transparent! gl-border-gray-100! gl-border-b-solid! gl-border-b-1!"
    role="columnheader"
    scope="col"
  >
    <div>
      Last used
    </div>
  </th>
  <th
    aria-colindex="5"
    class="gl-bg-transparent! gl-border-gray-100! gl-border-b-solid! gl-border-b-1!"
    role="columnheader"
    scope="col"
  >
    <div>
      Actions
    </div>
  </th>
</tr>
`;
